06.wc/my_wc.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = ARGV.getopts('lwc')
+  contents = read_contents
+  display_result(contents, options)
+end
+
+def display_result(contents, options)
+  total_lines = 0
+  total_words = 0
+  total_bytes = 0
+  contents.each do |file_name, content|
+    line_count = content.count
+    total_lines += line_count
+    word_count = content.join.split(/\s+/).count
+    total_words += word_count
+    byte_count = content.join.length
+    total_bytes += byte_count
+    puts format_line(line_count, word_count, byte_count, file_name, options)
+  end
+
+  return if contents.size == 1
+
+  puts format_line(total_lines, total_words, total_bytes, 'total', options)
+end
+
+def format_line(lines, words, bytes, file_or_total, options)
+  # lオプションのみの場合
+  # result = []
+  # result << format_number(lines)
+  # unless options['l']
+  #   result << format_number(words)
+  #   result << format_number(bytes)
+  # end
+  # result << " #{file_or_total}"
+  # result.join
+
+  # lオプション、wオプション、cオプションの場合
+  result = []
+  show_all = (options == { 'l' => false, 'w' => false, 'c' => false })
```

Comment:
この書き方だとlwc以外のオプションが増えたときに修正が必要になりますね。（あまりないと思いますが）
やりたいことは「valueが全部偽のときに全部表示」なので、

```suggestion
  show_all = options.values.none?
```

みたいな書き方ができます。

公式リファレンスでall?, any?, none? といったメソッドの説明を読んでみてください〜。

https://docs.ruby-lang.org/ja/latest/class/Enumerable.html